gn3/api/correlation.py

```diff
@@ -107,37 +112,29 @@ def __errors__(request_data, fields):
         if request_data is None:
             return ("No request data",)
 
-        return reduce(__field_errors__(args), fields, errors)
-
-    class OutputEncoder(json.JSONEncoder):
-        """
-        Class to encode output into JSON, for objects which the default
-        json.JSONEncoder class does not have default encoding for.
-        """
-        def default(self, o):
-            if isinstance(o, bytes):
-                return str(o, encoding="utf-8")
-            return json.JSONEncoder.default(self, o)
-
-    def __build_response__(data):
-        status_codes = {"error": 400, "not-found": 404, "success": 200}
-        response = make_response(
-            json.dumps(data, cls=OutputEncoder),
-            status_codes[data["status"]])
-        response.headers["Content-Type"] = "application/json"
-        return response
-
-    args = request.get_json()
+        return reduce(__field_errors__(request_data), fields, errors)
+
+    args = json.loads(request.get_json())
     request_errors = __errors__(
         args, ("primary_trait", "control_traits", "target_db", "method"))
     if request_errors:
-        return __build_response__({
+        return build_response({
             "status": "error",
             "messages": request_errors,
             "error_type": "Client Error"})
-    conn, _cursor_object = database_connector()
-    corr_results = partial_correlations_entry(
-        conn, trait_fullname(args["primary_trait"]),
-        tuple(trait_fullname(trait) for trait in args["control_traits"]),
-        args["method"], int(args.get("criteria", 500)), args["target_db"])
-    return __build_response__(corr_results)
+    return build_response({
+        "status": "success",
+        "results": queue_cmd(
+            conn=redis.Redis(),
+            cmd=compose_pcorrs_command(
+                trait_fullname(args["primary_trait"]),
+                tuple(
+                    trait_fullname(trait) for trait in args["control_traits"]),
+                args["method"], args["target_db"],
+                int(args.get("criteria", 500))),
+            job_queue=current_app.config.get("REDIS_JOB_QUEUE"),
+            env = {"PYTHONPATH": ":".join(sys.path), "SQL_URI": SQL_URI})})
+
+@correlation.route("/partial/<job_id>", methods=["GET"])
```

Comment:
+1 on the naming here :smile:

Reply:
Not sure what you are talking about, so I'll smugly pretend it wasn't an accident :smile: